sdk/storage/src/data_lake/properties.rs

```diff
@@ -47,6 +47,27 @@ impl<'a, 'b> AddAsHeader for Properties<'a, 'b> {
         // is safe since we know that comma is 1 byte in UTF8):
         builder.header(HEADER, &s[..s.len() - 1])
     }
+
+    fn add_as_header2(
+        &self,
+        request: &mut azure_core::Request,
+    ) -> Result<(), http::header::InvalidHeaderValue> {
+        // the header is a comma separated list of key=base64(value) see
+        // [https://docs.microsoft.com/rest/api/storageservices/datalakestoragegen2/filesystem/create#request-headers](https://docs.microsoft.com/rest/api/storageservices/datalakestoragegen2/filesystem/create#request-headers)
+        let mut s = String::new();
+        self.0.iter().for_each(|(k, v)| {
+            s.push_str(&format!("{}={},", k.as_ref(), base64::encode(v.as_ref())));
+        });
```

Comment:
I believe if you do the following you don't need to remove the last byte below. 
```suggestion
        let s = self.0.iter().map(|(k, v)| {
            format!("{}={}", k.as_ref(), base64::encode(v.as_ref()))
        }).collect::<Vec<_>>().join(",");
```

Reply:
I honestly blindly copied the preexisting code but your solution is brilliant! Love it ❤️!